File: scratch/check_espn_roster.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final client = HttpClient();
  try {
    final rosterUri = Uri.parse('https://site.api.espn.com/apis/site/v2/sports/soccer/fifa.world/teams/624/roster');
    final request = await client.getUrl(rosterUri);
    request.headers.set('User-Agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)');
    final response = await request.close();
    
    if (response.statusCode == 200) {
      final body = await response.transform(utf8.decoder).join();
      final data = jsonDecode(body) as Map<String, dynamic>;
      final athletes = data['athletes'] as List<dynamic>;
      if (athletes.isNotEmpty) {
        final firstAthlete = athletes[0] as Map<String, dynamic>;
        print('Athlete display name: ${firstAthlete['displayName']}');
        print('Athlete jersey: ${firstAthlete['jersey']}');
        print('Athlete age: ${firstAthlete['age']}');
        print('Athlete position map: ${firstAthlete['position']}');
        print('Athlete links: ${firstAthlete['links']}');
        
        // Check if there is an image URL in the athlete data. Usually it's at:
        // https://a.espncdn.com/i/headshots/soccer/players/full/{athleteId}.png
        final athleteId = firstAthlete['id'];
        print('Athlete Headshot URL: https://a.espncdn.com/i/headshots/soccer/players/full/$athleteId.png');
      }
    }
  } catch (e) {
    print('Error: $e');
  } finally {
    client.close();
  }
}

File: scratch/check_espn_roster.py
import requests
import json

# Test fetching teams from ESPN API
url_teams = "https://site.api.espn.com/apis/site/v2/sports/soccer/fifa.world/teams"
headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36',
    'Accept': 'application/json',
}

try:
    response = requests.get(url_teams, headers=headers, timeout=10)
    print(f"Teams status: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print("Teams root keys:", list(data.keys()))
        if 'sports' in data:
            sports = data['sports']
            print("Sports count:", len(sports))
            for sport in sports:
                print("  Sport:", sport.get('name'))
                leagues = sport.get('leagues', [])
                print("  Leagues count:", len(leagues))
                for league in leagues:
                    print("    League:", league.get('name'))
                    teams = league.get('teams', [])
                    print("    Teams count:", len(teams))
                    if teams:
                        first_team = teams[0]['team']
                        print("    First team keys:", list(first_team.keys()))
                        print("    First team ID:", first_team.get('id'))
                        print("    First team Name:", first_team.get('displayName'))
                        print("    First team Links:", first_team.get('links'))
    else:
        print("Failed to fetch teams:", response.text[:200])
except Exception as e:
    print(f"Error fetching teams: {e}")
